firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift

```diff
@@ -1745,12 +1778,24 @@ class BrowserViewController: UIViewController,
             tabManager.selectedTab?.loadRequest(PrivilegedRequest(url: homePanelURL) as URLRequest)
         }
         TelemetryWrapper.recordEvent(category: .action, method: .tap, object: .home)
+    }
 
-        guard isToolbarRefactorEnabled else { return }
-        let action = GeneralBrowserAction(navigateToHome: false,
-                                          windowUUID: windowUUID,
-                                          actionType: GeneralBrowserActionType.goToHomepage)
-        store.dispatch(action)
+    func didTapOnBack() {
+        // This code snippet addresses an issue related to navigation between pages in the same tab FXIOS-7309.
+        // Specifically, it checks if the URL bar is not currently focused (`!focusUrlBar`) and if it is
+        // operating in an overlay mode (`urlBar.inOverlayMode`).
+        dismissUrlBar()
+        updateZoomPageBarVisibility(visible: false)
+        tabManager.selectedTab?.goBack()
+    }
+
+    func didTapOnForward() {
+        // This code snippet addresses an issue related to navigation between pages in the same tab FXIOS-7309.
+        // Specifically, it checks if the URL bar is not currently focused (`!focusUrlBar`) and if it is
+        // operating in an overlay mode (`urlBar.inOverlayMode`).
+        dismissUrlBar()
+        updateZoomPageBarVisibility(visible: false)
```

Comment:
nit: can move to its on method or create a general method to avoid repetitive code